module/wishlist/helper.js

```diff
@@ -0,0 +1,185 @@
+const userExtensionQueries = require(DB_QUERY_BASE_PATH + '/userExtension')
+const projectTemplateQueries = require(DB_QUERY_BASE_PATH + '/projectTemplates')
+
+module.exports = class UserExtensioHelper {
+	/**
+	 * add wishlist
+	 * @method
+	 * @name add
+	 * @param {projectTempleteId} - projectTempleteId
+	 * @param {userId} -            userId
+	 * @returns {Object} .
+	 */
+	static async add(projectTempleteId, userId) {
+		try {
+			let wishlistItem = {
+				_id: projectTempleteId,
+				createdAt: new Date(),
+			}
+
+			// Find the userExtension document for the given userId
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			let updateuserExtensionDocument
+			if (userExtensionDocument && userExtensionDocument.length > 0) {
+				// If the document exists, update the wishlist by appending the new item
+				updateuserExtensionDocument = await userExtensionQueries.findAndUpdate(
+					{ userId },
+					{ $addToSet: { wishlist: wishlistItem } }
+				)
+			} else {
+				// If no document exists, create a new one
+				const newUserExtension = {
+					userId,
+					wishlist: [wishlistItem],
+				}
+				updateuserExtensionDocument = await userExtensionQueries.createUserExtension(newUserExtension)
+			}
+			if (updateuserExtensionDocument) {
+				return {
+					success: true,
+					message: CONSTANTS.apiResponses.WISHLIST_ADDED,
+				}
+			} else {
+				throw {
+					message: CONSTANTS.apiResponses.WISHLIST_NOT_ADDED,
+					status: HTTP_STATUS_CODE.bad_request.status,
+				}
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * remove wishlist
+	 * @method
+	 * @name remove
+	 * @param {projectTempleteId} - pprojectTempleteId
+	 * @param {userId} -            userId
+	 * @returns {Object} .
+	 */
+	static async remove(projectTempleteId, userId) {
+		try {
+			// Find the userExtension document for the given userId
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			if (userExtensionDocument) {
+				// If the document exists, update the wishlist by removing the specified item
+				let updatedDocument = await userExtensionQueries.findAndUpdate(
+					{ userId, wishlist: { $elemMatch: { _id: projectTempleteId } } },
+					{ $pull: { wishlist: { _id: projectTempleteId } } },
+					{ new: true }
+				)
+				if (updatedDocument) {
+					return {
+						success: true,
+						message: CONSTANTS.apiResponses.WISHLIST_REMOVED,
+					}
+				} else {
+					throw {
+						message: CONSTANTS.apiResponses.PROJECTTEMPLATES_NOTFOUND,
+						status: HTTP_STATUS_CODE.bad_request.status,
+					}
+				}
+			} else {
+				throw {
+					message: CONSTANTS.apiResponses.USEREXTENSION_NOTFOUND,
```

Comment:
@praveenKDass if user extension is not there ,immediately throw error. move this to line 70. if no error caut which means data exist,so continue the logic. like this we can avoid multiple if else

Reply:
Added @VISHNUDAS-tunerlabs